File: src/tests/fidl/source_compatibility/table-member-remove/llcpp/step_01_after.cc
// Copyright 2021 The Fuchsia Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include <fidl/fidl.test.tablememberremove/cpp/wire.h>  // nogncheck
namespace fidl_test = fidl_test_tablememberremove;

// [START contents]
void use_table(const fidl_test::wire::Profile& profile) {
  if (profile.has_timezone()) {
    printf("timezone: %s", profile.timezone().data());
  }
  if (profile.has_temperature_unit()) {
    printf("preferred unit: %s", profile.temperature_unit().data());
  }
}
// [END contents]

int main(int argc, const char** argv) { return 0; }

File: src/tests/fidl/source_compatibility/table-member-remove/dart/step_01_after.dart
// Copyright 2021 The Fuchsia Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

// TODO(https://fxbug.dev/84961): Fix null safety and remove this language version.
// @dart=2.9

import 'package:fidl_fidl_test_tablememberremove/fidl_async.dart' as fidllib;

// [START contents]
void useTable(fidllib.Profile profile) {
  if (profile.timezone != null) {
    print('timezone: ${profile.timezone}');
  }
  if (profile.temperatureUnit != null) {
    print('preferred unit: ${profile.temperatureUnit}');
  }
  profile.$unknownData?.forEach((ordinal, data) {
    print('unknown ordinal $ordinal with bytes ${data.data}');
  });
}
// [END contents]
